src/librustc/middle/dead.rs

```diff
@@ -464,21 +460,19 @@ impl<'a, 'tcx> DeadVisitor<'a, 'tcx> {
         // method of a private type is used, but the type itself is never
         // called directly.
         let impl_items = self.tcx.impl_items.borrow();
-        match self.tcx.inherent_impls.borrow().get(&self.tcx.map.local_def_id(id)) {
-            None => (),
-            Some(impl_list) => {
+        if let Some(impl_list) =
+            self.tcx.inherent_impls.borrow().get(&self.tcx.map.local_def_id(id)) {
                 for impl_did in impl_list.iter() {
                     for item_did in impl_items.get(impl_did).unwrap().iter() {
                         if let Some(item_node_id) =
-                                self.tcx.map.as_local_node_id(item_did.def_id()) {
-                            if self.live_symbols.contains(&item_node_id) {
-                                return true;
+                            self.tcx.map.as_local_node_id(item_did.def_id()) {
+                                if self.live_symbols.contains(&item_node_id) {
+                                    return true;
+                                }
                             }
```

Comment:
Same here -- this is OK the way it was, or it could be:
```rust
                        if let Some(item_node_id) =
                            self.tcx.map.as_local_node_id(item_did.def_id()) {
                            if self.live_symbols.contains(&item_node_id) {
                                return true;
                            }
                        }
```